### PR TITLE
Benchmarks: added Go, abs2 in mandelbrot, moved plot to front page

### PR DESCRIPTION
**Pull request by johnfgibson**
This PR

  * Adds benchmark results from Go 
  * Switches to a uniform user-defined function `abs2` within the mandelbrot benchmark and renames that `userfunc_mandelbrot`. 
  * Folds in revised results from a small `range(t)` fix in Python benchmarks [JuliaLang/julia#24089](https://github.com/JuliaLang/julia/pull/24089)
  * Swaps the placement of the benchmark plot (now on front page) and table (now behind a link)
  * Revises front-page text describing the aim of the benchmark, trying to highlight the focus on identical algorithms and code patterns instead of fastest method of computing the result.

### CODE REVIEW

File: _includes/benchmarks.html [hunk 1]
```diff
@@ -1,30 +1,31 @@
 <table class="benchmarks">
 <colgroup>
-<col class="name">
-<col class="relative" span="11">
+<col class="name"></col>
+<col class="relative" span="12"></col>
```
Comment (StefanKarpinski) [inline _includes/benchmarks.html]: Can we get these without the `</col>` – browsers render them in the output? I made that change to Julia master already, so if you rebase your changes, the Perl script that generates this should not include these.

**Comment by ViralBShah** [comment]
Are all the other benchmarks using `abs2` uniformly. I only see the julia change in here, and am assuming that we may have needed it in other cases too.

**Comment by KristofferC** [comment]
The actual benchmark code is in the Julia repo though, right?

**Comment by johnfgibson** [comment]
@ViralBShah: I'm working on a PR to JuliaLang/julia with the change to `abs2` and name `userfunc_mandelbrot` in all the benchmark codes, probably will file tomorrow.

**Comment by ViralBShah** [comment]
Thank you. Forgot for a moment that these are separate repos.

**Comment by ChrisRackauckas** [comment]
Anything blocking this?

**Comment by StefanKarpinski** [comment]
I wasn't sure if we should merge this before a PR exists for the corresponding code changes, but sure, no harm I guess.

**Pull request merged**